ksql-execution/src/main/java/io/confluent/ksql/execution/builder/KsqlQueryBuilder.java

```diff
@@ -135,6 +136,28 @@ public KsqlQueryBuilder withKsqlConfig(final KsqlConfig newConfig) {
         .push(context);
   }
 
+  @SuppressWarnings("unchecked")
+  public KeySerde<Object> buildKeySerde(
```

Comment:
The generics on this call are wrong.  It's a pervasive anti-pattern within the code base, but it's wrong. Returning `KeySerde<Object>` is saying its a key serde that can handle `Object`s, but it can't. The returned serde can only handle _either_ `Struct` or `Windowed<Struct>`.  The fact that you're having to first cast to `(KeySerde)` is a clear indication of this: `KeySerde<Struct>` can not be cast to `KeySerde<Object>` because they are not compatible types.

This is why there are two methods for explicitly handling windowed vs non-windowed serde.

The correct return type for this method is `KeySerde<?>` i.e. a key serde of an unknown type.  This of course gives up type safety, which we should avoid if we can.

The following requires no casts and generates no warnings, though it will of course require other code to change:

```java
   public KeySerde<?> buildKeySerde(
      final KeyFormat keyFormat,
      final PhysicalSchema physicalSchema,
      final QueryContext queryContext
  ) {
    if (keyFormat.getWindowInfo().isPresent()) {
      return buildKeySerde(
          keyFormat.getFormatInfo(),
          keyFormat.getWindowInfo().get(),
          physicalSchema,
          queryContext
      );
    }

    return buildKeySerde(
        keyFormat.getFormatInfo(),
        physicalSchema,
        queryContext
    );
  }
```

Unfortunately, the generics on our key type are... foobared, and will require some effort to correct and give us type safety.  When introducing new functionality or refactoring existing, we should endeavour to make it type safe and then us hacky casts at the point this interacts with old code.